mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+    x_low = cwt(data, Ws, use_fft=False).ravel()
+
+    phases = np.angle(x_low)
+    trigger_inds, _ = peak_finder(phases)
+
+    n_samples = data.size
+    n_freqs = len(freqs)
+
+    if tmin >= tmax:
+        raise ValueError('tmin has to be smaller than tmax')
+
+    # obtain min and max indices
+    n_times_min = -1 * int(round(tmin * sfreq))
+    n_times_max = int(round(tmax * sfreq))
+    times = np.arange(-n_times_min, n_times_max, dtype=np.float) / sfreq
+
+    # makes sure all triggers allow full epoch, get rid of those that don't:
+    trigger_inds = trigger_inds[trigger_inds > n_times_min]
+    trigger_inds = trigger_inds[trigger_inds < n_samples - n_times_max]
+
+    # Compute filtered data
+    Ws = morlet(sfreq, freqs, n_cycles=n_cycles, sigma=None,
+                zero_mean=True)
+
+    parallel, my_abs_cwt, n_jobs = parallel_func(_abs_cwt, n_jobs)
+
+    if n_jobs == 1:
+        ampmat = np.empty((n_freqs, n_samples), dtype=np.float)
+        for e, W in enumerate(Ws):
+            ampmat[e] = np.array(_abs_cwt(data, W))
+    else:
+        ampmat = np.array(parallel(my_abs_cwt(data, W) for W in Ws))
+
+    # zscore for normalization
+    ampmat -= np.mean(ampmat, axis=1)[:, None]
+    ampmat /= np.std(ampmat, axis=1)[:, None]
+
+    # phase-triggered average of raw signal:
+    # and phase-triggered time-frequency amplitude values (normalized):
+    avg = np.zeros(times.size)
+    traces = np.zeros((n_freqs, times.size), dtype=np.float)
+    for ind in trigger_inds:
+        avg += data[0, ind - n_times_min: ind + n_times_max]
+        traces += ampmat[:, ind - n_times_min: ind + n_times_max]
+    avg /= len(trigger_inds)
+    traces /= len(trigger_inds)
+
+    return times, ampmat, traces, avg, trigger_inds
+
+
+def compute_cfc_stats(sfreq, ampmat, traces, freqs, trigger_inds,
+                      n_surrogates=10000, alpha=0.001, random_state=None):
+    """ Stats function for cross frequency coupling.
+
+    Parameters
+    ----------
+    sfreq : float
+        Sampling frequency.
+    ampmat : ndarray, shape (nfreq, n_samples)
+        Amplitude traces.
+    traces : ndarray, shape (n_epochs, n_times)
+        Normalized amplitude traces.
+    freqs : array, shape (n_freqs,)
+        Frequency points across range at which amplitudes are computed.
+    trigger_inds : array (trigger_inds,)
+        Indices of maxima in the phase signal.
+    n_surrogates : int
+        Number of surrogates. Default 10 ** 4.
+    alpha : float
+        Error rate allowed. Default 0.001.
+    random_state : None | int | instance of np.random.RandomState
+        np.random.RandomState to initialize the noise estimation.
+        Default value is None.
+
+    Returns
+    -------
+    ztraces : ndarray, shape (n_freqs, n_times)
+        Z-scored amplitude traces.
+    z_threshold : float
+        FDR corrected threshold for z-scored amplitude traces.
+    """
+
+    n_freqs = len(freqs)
+    n_samples = ampmat.size / n_freqs
```

Comment:
Maybe just my opinion, but I don't like using "size" all that much, because it reveals less information about the object that it's attached to. E.g., you don't know if it's a 1-d or 2-d object which can be confusing. In the case of ampmat, this is a 2-d matrix of shape (n_freqs, n_times), so rather than using `n_samples = ampmat.size / n_freqs` it's much clearer to use `n_samples = ampmat.shape[1]`. No?